OMCompiler/Compiler/BackEnd/BackendDAEUtil.mo

```diff
@@ -4390,31 +4390,36 @@ protected
   BackendDAE.Equation eq;
   BackendDAE.VarKind varKind;
   DAE.ComponentRef cr;
+  String eqnstr;
   constant Boolean debug = false;
 algorithm
   // Get equation and variable from integer index
   try
     BackendDAE.EQSYSTEM(orderedVars=vars, orderedEqs=eqns) := isyst;
     BackendDAE.SHARED(globalKnownVars=globalKnownVars) := ishared;
+
+    eq := BackendEquation.get(eqns, inEquation);
+    BackendDAE.VAR(varName = cr , varKind = varKind) := BackendVariable.getVarAt(vars, inVariable);
+    // Compute solvability
+    solvability := match(varKind)
+      // Case variable is a state
+      case(BackendDAE.STATE()) algorithm
```

Comment:
Why do we need this here? Minimal tearing should only try to match discrete vars and they are never continuous states. Or is this function generally used?

Reply:
@AnHeuermann @kabdelhak I am sorry for this but this function is now obsolete and will be removed. Minimal tearing does not use it anymore as of #733. I will incorporate the improved error messages to that PR.
 
Why was it needed before anyway? The adjacency matrices have solvability information in one way or another. We should try to use that information.  Is there something specific this function was looking for?

Reply:
I'm fine with it being removed completely.
I think I wanted to see if this makes a difference for the algorithms that were not handled correctly by minimal tearing and didn't remove it for this PR.

Reply:
Sounds good, this was implemented by a student so he probably was not aware of available information and just created it by himself.

Reply:
Ah one second. 

I added a mapping to the EqSystem structure `type AdjacencyMatrixMapping = tuple<array<list<Integer>>, array<Integer>, IndexType, Boolean, Boolean>` You can check the `IndexType`, it should be `BackendDAE.SOLVABLE()` otherwise you are operating on wrong information if you use the adjacency matrix. I am not sure if this is done by default, but if you don't update the adjacency matrix with solvability restrictions after matching, it only yields information about general occurrence and not solvability.

Reply:
@kabdelhak That's actually good to have. It would have been better if the matrix type was a part of the adjacency matrix itself. But since the matrix is now just a type definition 
type AdjacencyMatrix = array<AdjacencyMatrixElement>
and not really a structure it will be a lot of work to add it there. 

While we are at it I also noticed that the  BackendDAE.ABSOLUTE() type of adjacency does not specify if it was _solvable only_ or _all occurrence_. It should be changed to something like BackendDAE.ABSOLUTE(boolean solvable) or split in to two BackendDAE.ABSOLUTE_NORMAL() and BackendDAE.ABSOLUTE_SOLVABLE(). There are places where we check if a matrix is solvable, but we have already taken the solvable matrix and converted it to absolute (but still only solvable entries).

e.g. `Matching.hasIndexTypeSolvableAndUnprocessedScalar()` checks for solvable in some cases but the solvable is already converted to absolute in `Causalize.singularSystemCheck1()` but still with only solvable entries. Maybe this is a false positive but it is worth a second look.

Anyway I think we should avoid the normal/solved adjacency matrix **all together** and use the enhanced one exclusively.